portfolio/src/main/webapp/comments.js

```diff
@@ -12,16 +12,84 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function loadComments(){
-  const maxComments = parseInt(document.getElementById("max-comments-select").value);
-  debugLog(maxComments);
-  fetch("/data" + "?max_comments=" + maxComments).then(response => response.json()).then((comments) => {
-    debugLog(comments);
-    const commentsList = document.getElementById('comments-container');
-    commentsList.innerHTML = '';
-    comments.forEach((comment) => {
-      commentsList.appendChild(createCommentElement(comment.email, comment.comment, comment.date));
-    });
+function loadCommentsPage(pgNumber = 1){
+  const commentsPerPage = parseInt(document.getElementById('comments-per-page-select').value);
+  const DOM_WAIT_TIME = 750;
+  debugLog("commentsPerPage=" + commentsPerPage);
+  fetch("/data" + "?comments_per_page=" + commentsPerPage+ "&pg_number=" + pgNumber).then(response => response.json()).then((response) => {
+    renderComments(response.comments);
+    renderPagination(response.lastPage, pgNumber);
+  });
+  
+  // need to wait so that the pagination renders and document.getElementById("pg" + pgNumber) doesn't return null
+  setTimeout(function(){
+    addClass(document.getElementById("pg" + pgNumber), "active");
```

Comment:
Can this be moved into renderPagination to avoid the setTimeout? Suggestion is based on the comment you added above.

Reply:
remodeled the process of finding the button that should be activated